### PR TITLE
[#3805] improvement(CI): upload logs of catalog and add free up disk space

### PR DESCRIPTION
**Pull request by zhoukangcn**
### What changes were proposed in this pull request?

- upload logs of catalog
- add free up disk space

### Why are the changes needed?

Fix: #3805

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

github workflow

### CODE REVIEW

**Comment by zhoukangcn** [comment]
@yuqi1129 Could you help to review this pr when you have time?

log is uploaded in  https://github.com/datastrato/gravitino/actions/runs/9402207448?pr=3806

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -89,6 +89,10 @@ jobs:
       - name: Test publish to local
         run: ./gradlew publishToMavenLocal -x test -PjdkVersion=${{ matrix.java-version }}
 
+      - name: Free up disk space
+        run: |
+          dev/ci/util_free_space.sh
```
Comment (yuqi1129) [inline .github/workflows/build.yml:94]: The reason you added this script is that the failed case you pasted has run out of disk space. Am I right?

**Comment by yuqi1129** [comment]
LGTM, good fix

File: .github/workflows/build.yml (hunk 1 above)
Comment (zhoukangcn) [reply .github/workflows/build.yml:94]: Yes. 
```
Disk usage:
Filesystem      Size  Used Avail Use% Mounted on
overlay          73G   68G  4.9G  94% /
```
```
ERROR: Doris FE (version < 2.1.0) can not start with less than 6G disk space.
```

Comment (yuqi1129) [reply .github/workflows/build.yml:94]: I see.

**Pull request merged**